File: dongmin/week5/k번째 수.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];

        List<Integer> arr = new ArrayList<>();
        for (int i = 0; i < commands.length; i++) {
            for(int j=(commands[i][0]-1);j<commands[i][1];j++){     //첫번째 인덱스에 접근하기 위해 -1을 빼준다.
                arr.add(array[j]);      //값을 arr에 넣어준다.
            }
            Collections.sort(arr);      //arr에 들어간 값을 sort 메서드를 통해 오름차순 정렬한다.
            answer[i]=arr.get((commands[i][2])-1);      //n번째 인덱스에 들어가 그 값을 answer에 i번째에 넣어준다.
            arr.clear();                //arr을 다시 초기화하고 commands에 다음 원소에 접근한다.
        }

        return answer;
    }
}

File: yongkeun/5week/K번째수.js
function solution(array, commands) {
  var answer = [];
  for (let i = 0; i < commands.length; i++) {
    //commands 만큼 반복
    a = array.slice(commands[i][0] - 1, commands[i][1]).sort((a, b) => a - b); //2번째부터 5번째까지라고 문제지에 적혀있는데 4번째이므로 index값 1~4을 짤라야 함 자르기 위해 slice를 이용해 자르고, 오름차순 정렬까지 한번에 하여 a라는 변수에 할당
    answer.push(a[commands[i][2] - 1]); //배열에 push하는데 세 번째 숫자를 뽑아낼때 index는 2인 숫자를 뽑아내므로 -1
  }
  return answer;
}
